interactions-transitions-assignment/part-1-bar-chart/us-emissions-interaction.js

```diff
@@ -21,11 +22,138 @@ function createChart(elementId) {
+        .call(xAxis)
+        .selectAll("text")
+        .attr("transform", "rotate(-45)")
+        .style("text-anchor", "end");
+    // y axis
+    g.append('g')
+        .attr("transform", `translate(0,0)`)
+        .call(yAxis);
+    // labels for x and y axis and graph title
+    g.append("text")
+        .attr("x", innerWidth / 2.5)
+        .attr("y", innerHeight - margins.bottom + 100)
+        .attr("text-anchor", "middle")
+        .style("font-size", "16px")
+        .text("States");
+    
+      g.append("text")
+        .attr("transform", "rotate(-90)")
+        .attr("y", 0 - margins.left + 12)
+        .attr("x", 0 - (innerHeight / 2))
+        .attr("dy", "0.05em")
+        .style("text-anchor", "middle")
+        .style("font-size", "16px")
+        .text("Emissions");
+    
+    g.append("text")
+      .attr("x", innerWidth / 2.5)
+      .attr("y", margins.top / 100)
+      .attr("text-anchor", "middle")
+      .style("font-size", "20px")
+      .text("Emissions by State ");
+    
+    // legend for regions
+    const legend = svg.append("g")
+      .attr("transform", `translate(${margins.left + 120},${margins.top})`);
+    const regions = Array.from(new Set(data.map(d => d.Region)));
+    regions.forEach((region, i) => {
+      const legendRow = legend.append("g")
+        .attr("transform", `translate(0,${i * 20})`); 
+      legendRow.append("rect")
+        .attr("width", 10)
+        .attr("height", 10)
+        .attr("fill", colorScale(region));
+  
+      legendRow.append("text")
+        .attr("x", -10)
+        .attr("y", 10)
+        .attr("text-anchor", "end")
+        .style("text-transform", "capitalize")
+        .text(region);
     });
-}
+    // create bars
+    let bars = g.selectAll(".bar")
+        .data(data)
+        .join("rect")
+        .attr("class", "bar");
+
+  // add event listeners for the radio buttons
+    d3.selectAll('input[name="sort"]').on('change', function() {
+        let sortOrder;
+        if (this.value === 'state') {
+        sortOrder = (a, b) => d3.ascending(a.State, b.State);
+        } else if (this.value === 'emissions-desc') {
+        sortOrder = (a, b) => d3.descending(a.Emissions, b.Emissions);
+        } else if (this.value === 'emissions-asc') {
+        sortOrder = (a, b) => d3.ascending(a.Emissions, b.Emissions);
+        }
+
+        // sort data
+        data.sort(sortOrder);
+
+        // update scales
+        stateScale.domain(data.map(d => d.State));
+        emissionsScale.domain([0, d3.max(data, d => d.Emissions)]);
+
+        // transition bars
+        bars.transition()
+        .duration(1500)
+        .delay((_d, i) => i * Number(d3.select('#delay').node().value))
+        .attr("x", d => stateScale(d.State))
+        .attr("y", d => emissionsScale(d.Emissions))
+        .attr("height", d => innerHeight - emissionsScale(d.Emissions));
+
+        g.select('.x.axis')
+            .transition()
+            .duration(1500)
```

Comment:
I would add a delay to the transition for the axis as well so that the bars and the axis move together